### PR TITLE
Empty shortcuts

### PR DESCRIPTION
**Pull request by forki**
more shortcuts in style of #5348

### CODE REVIEW

File: src/fsharp/FSharp.Core/array.fs [hunk 1]
```diff
@@ -278,6 +281,9 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("DistinctBy")>]
         let distinctBy projection (array:'T[]) =
+            let length = array.Length
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/array.fs]: @forki   --- Null check here:  Using array without verifying if it is not null.

**Comment by forki** [comment]
Looks like we are missing a test then. I'll add both.

Kevin Ransom (msft) <notifications@github.com> schrieb am Do., 9. Aug.
2018, 20:35:

File: src/fsharp/FSharp.Core/array.fs (hunk 1 above)
Comment (forki) [reply src/fsharp/FSharp.Core/array.fs]: just looked it up - the null check is already in outer function

Comment (forki) [reply src/fsharp/FSharp.Core/array.fs]: so I think it's fine

Comment (KevinRansom) [reply src/fsharp/FSharp.Core/array.fs]: I believe it's a public API, I'm pretty sure a null check is required,

**Comment by KevinRansom** [comment]
The VSTS: public -CI is the gatekeeper now.  Fails on ci_part2 and ci_part3.

From ci_part2:

````
Errors and Failures
2018-08-09T08:10:25.2469989Z 
2018-08-09T08:10:25.2470267Z 1) Failed : FSharp.Core.UnitTests.FSharp_Core.Microsoft_FSharp_Collections.ArrayModule.distinctBy
2018-08-09T08:10:25.2470608Z Expected System.ArgumentNullException exception, got: System.NullReferenceException: Object reference not set to an instance of an object.
2018-08-09T08:10:25.2470957Z    at Microsoft.FSharp.Collections.ArrayModule.DistinctBy[T,TKey](FSharpFunc`2 projection, T[] array) in D:\a\1\s\src\fsharp\FSharp.Core\array.fs:line 284
2018-08-09T08:10:25.2471350Z    at <StartupCode$FSharp-Core-UnitTests>.$ArrayModule.distinctBy@247-1.Invoke(Unit unitVar0) in D:\a\1\s\tests\FSharp.Core.UnitTests\FSharp.Core\Microsoft.FSharp.Collections\ArrayModule.fs:line 247
2018-08-09T08:10:25.2471716Z    at FSharp.Core.UnitTests.LibraryTestFx.CheckThrowsExn[a](FSharpFunc`2 f) in D:\a\1\s\tests\FSharp.Core.UnitTests\LibraryTestFx.fs:line 17
2018-08-09T08:10:25.2472105Z at FSharp.Core.UnitTests.LibraryTestFx.CheckThrowsExn[a](FSharpFunc`2 f) in D:\a\1\s\tests\FSharp.Core.UnitTests\LibraryTestFx.fs:line 21
2018-08-09T08:10:25.2472789Z at FSharp.Core.UnitTests.FSharp_Core.Microsoft_FSharp_Collections.ArrayModule.distinctBy() in D:\a\1\s\tests\FSharp.Core.UnitTests\FSharp.Core\Microsoft.FSharp.Collections\ArrayModule.fs:line 250
2018-08-09T08:10:25.2473051Z 
2018-08-09T08:10:25.2506966Z Run Settings
2018-08-09T08:10:25.2508612Z     RuntimeFramework: V4.0
2018-08-09T08:10:25.2509181Z     WorkDirectory: D:\a\1\s\release\net40\bin
2018-08-09T08:10:25.2509393Z     Verbose: True
2018-08-09T08:10:25.2509621Z     NumberOfTestWorkers: 2
2018-08-09T08:10:25.2509759Z 
2018-08-09T08:10:25.2521620Z Test Run Summary
2018-08-09T08:10:25.2521958Z     Overall result: Failed
2018-08-09T08:10:25.2532180Z    Tests run: 5389, Passed: 5388, Errors: 0, Failures: 1, Inconclusive: 0
2018-08-09T08:10:25.2536456Z      Not run: 6, Invalid: 0, Ignored: 3, Explicit: 3, Skipped: 0
2018-08-09T08:10:25.2536878Z   Start time: 2018-08-09 08:05:58Z
2018-08-09T08:10:25.2537276Z     End time: 2018-08-09 08:10:25Z
2018-08-09T08:10:25.2540232Z     Duration: 266.925 seconds
````

From ci_part3
````
2018-08-09T07:39:18.3589411Z Errors and Failures
2018-08-09T07:39:18.3598695Z 
2018-08-09T07:39:18.3599065Z 1) Failed : FSharp.Core.UnitTests.FSharp_Core.Microsoft_FSharp_Collections.ArrayModule.distinctBy
2018-08-09T07:39:18.3599355Z Expected System.ArgumentNullException exception, got: System.NullReferenceException: Object reference not set to an instance of an object.
2018-08-09T07:39:18.3599698Z    at Microsoft.FSharp.Collections.ArrayModule.DistinctBy[T,TKey](FSharpFunc`2 projection, T[] array)
2018-08-09T07:39:18.3600032Z    at <StartupCode$FSharp-Core-UnitTests>.$ArrayModule.distinctBy@247-1.Invoke(Unit unitVar0) in D:\a\1\s\tests\FSharp.Core.UnitTests\FSharp.Core\Microsoft.FSharp.Collections\ArrayModule.fs:line 247
2018-08-09T07:39:18.3600419Z    at FSharp.Core.UnitTests.LibraryTestFx.CheckThrowsExn[a](FSharpFunc`2 f) in D:\a\1\s\tests\FSharp.Core.UnitTests\LibraryTestFx.fs:line 17
2018-08-09T07:39:18.3600973Z at FSharp.Core.UnitTests.LibraryTestFx.CheckThrowsExn[a](FSharpFunc`2 f) in D:\a\1\s\tests\FSharp.Core.UnitTests\LibraryTestFx.fs:line 21
2018-08-09T07:39:18.3601335Z at FSharp.Core.UnitTests.FSharp_Core.Microsoft_FSharp_Collections.ArrayModule.distinctBy() in D:\a\1\s\tests\FSharp.Core.UnitTests\FSharp.Core\Microsoft.FSharp.Collections\ArrayModule.fs:line 250
2018-08-09T07:39:18.3601582Z 
2018-08-09T07:39:18.3607607Z Run Settings
2018-08-09T07:39:18.3611436Z     Internal Trace: Off
2018-08-09T07:39:18.3611616Z 
2018-08-09T07:39:18.3627692Z Test Run Summary
2018-08-09T07:39:18.3629374Z   Overall result: Failed
2018-08-09T07:39:18.3656542Z   Test Count: 5390, Passed: 5384, Failed: 1, Inconclusive: 0, Skipped: 5
2018-08-09T07:39:18.3656870Z     Failed Tests - Failures: 1, Errors: 0, Invalid: 0
2018-08-09T07:39:18.3657088Z     Skipped Tests - Ignored: 3, Explicit: 2, Other: 0
2018-08-09T07:39:18.3657615Z   Start time: 2018-08-09 07:35:13Z
2018-08-09T07:39:18.3657842Z     End time: 2018-08-09 07:39:18Z
2018-08-09T07:39:18.3658024Z     Duration: 244.700 seconds
2018-08-09T07:39:18.3658158Z 
2018-08-09T07:39:18.3823079Z -----------------------------------------------------------------
2018-08-09T07:39:18.3835537Z Error: Running tests coreclr-coreunit failed, see logs above-- FAILED
2018-08-09T07:39:18.3835693Z -----------------------------------------------------------------
2018-08-09T07:39:19.4902600Z ##[error]Cmd.exe exited with code '1'.

````

File: src/fsharp/FSharp.Core/array.fs (hunk 1 above)
Comment (cartermp) [reply src/fsharp/FSharp.Core/array.fs]: Yes, this is a public API. Prior to this change the null check was the first line to this function.

Comment (forki) [reply src/fsharp/FSharp.Core/array.fs]: oh was looking on the wrong one

Comment (forki) [reply src/fsharp/FSharp.Core/array.fs]: fixed. and thanks!

**Comment by KevinRansom** [comment]
@forki 
Thanks mate

**Pull request merged**